infrastructure/TuDa.CIMS.MigrationService/Worker.cs

```diff
@@ -0,0 +1,142 @@
+) : BackgroundService
+{
+    public const string ActivitySourceName = "Migrations";
+    private static readonly ActivitySource s_activitySource = new(ActivitySourceName);
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        using var activity = s_activitySource.StartActivity(
+            "Migrating database",
+            ActivityKind.Client
+        );
+
+        try
+        {
+            using var scope = serviceProvider.CreateScope();
+            var dbContext = scope.ServiceProvider.GetRequiredService<CIMSDbContext>();
+
+            await EnsureDatabaseAsync(dbContext, stoppingToken);
+            await RunMigrationAsync(dbContext, stoppingToken);
+            await SeedDataAsync(dbContext, stoppingToken);
+        }
+        catch (Exception ex)
+        {
+            activity?.RecordException(ex);
+            throw;
+        }
+
+        hostApplicationLifetime.StopApplication();
+    }
+
+    private static async Task EnsureDatabaseAsync(
+        CIMSDbContext dbContext,
+        CancellationToken cancellationToken
+    )
+    {
+        var dbCreator = dbContext.GetService<IRelationalDatabaseCreator>();
+
+        var strategy = dbContext.Database.CreateExecutionStrategy();
+        await strategy.ExecuteAsync(async () =>
+        {
+            // Create the database if it does not exist.
+            // Do this first so there is then a database to start a transaction against.
+            if (!await dbCreator.ExistsAsync(cancellationToken))
+            {
+                await dbCreator.CreateAsync(cancellationToken);
+            }
+        });
+    }
+
+    private static async Task RunMigrationAsync(
+        CIMSDbContext dbContext,
+        CancellationToken cancellationToken
+    )
+    {
+        var strategy = dbContext.Database.CreateExecutionStrategy();
+        await strategy.ExecuteAsync(async () =>
+        {
+            await dbContext.Database.MigrateAsync(cancellationToken);
+        });
+    }
+
+    private async Task SeedDataAsync(CIMSDbContext dbContext, CancellationToken cancellationToken)
+    {
+        var hazard = new Hazard { Id = Guid.NewGuid(), Name = "Hazard" };
+        var room = new Room { Id = Guid.NewGuid(), Name = "Haupt" };
+
+        var chemical = new Chemical
+        {
+            Cas = "12-33-34232",
+            Unit = "Liter",
+            Id = Guid.NewGuid(),
+            Room = room,
+            Name = "Chemikalie",
+            ItemNumber = "12345",
+            Shop = "Chemikalien Shop",
+            Hazards = [hazard],
+            Note = "Notiz",
+        };
+
+        var consumable = new Consumable
+        {
+            Manufacturer = "Glasmacher",
+            SerialNumber = "12343842738437293",
+            Id = Guid.NewGuid(),
+            Room = room,
+            Name = "Verbrauchsgegenstand",
+            ItemNumber = "12345",
+            Shop = "Glas Shop",
+            Note = "Notiz",
+        };
```

Comment:
Do the example values need to be removed later?

Reply:
When the Database is completely empty, this method creates some data to seed the database with it.
So no, they don't have to be removed later.